src/scripts/treelite_python/compile.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""
+TreeLite/Python inferencing script
+"""
+import os
+import sys
+import argparse
+import logging
+import numpy
+from distutils.util import strtobool
+import pandas as pd
+import treelite, treelite_runtime
+
+# Add the right path to PYTHONPATH
+# so that you can import from common.*
+COMMON_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", ".."))
+
+if COMMON_ROOT not in sys.path:
+    print(f"Adding {COMMON_ROOT} to PYTHONPATH")
+    sys.path.append(str(COMMON_ROOT))
+
+# useful imports from common
+from common.metrics import MetricsLogger
+from common.io import input_file_path
+
+
+def get_arg_parser(parser=None):
+    """Adds component/module arguments to a given argument parser.
+
+    Args:
+        parser (argparse.ArgumentParser): an argument parser instance
+
+    Returns:
+        ArgumentParser: the argument parser instance
+
+    Notes:
+        if parser is None, creates a new parser instance
+    """
+    # add arguments that are specific to the script
+    if parser is None:
+        parser = argparse.ArgumentParser(__doc__)
+
+    group_i = parser.add_argument_group("Input Data")
+    group_i.add_argument("--data",
```

Comment:
you do not need the --data argument at all for this script

Reply:
same for outputs

Reply:
OK I will update the PR to reflect this. I was wondering about where the .so should be